File: leetcode/editor/cn/258_各位相加_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func Test_givenNormal_when258_thenSuccess(t *testing.T) {
	assert.Equal(t, 0, addDigits(0))
	assert.Equal(t, 1, addDigits(1))
	assert.Equal(t, 2, addDigits(38))
}

//leetcode submit region begin(Prohibit modification and deletion)
func addDigits(num int) int {
	return (num-1)%9 + 1
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/258_各位相加_test.py
import unittest


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        self.assertEqual(2, self.solution.addDigits(38))
        self.assertEqual(0, self.solution.addDigits(0))


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def addDigits(self, num: int) -> int:
        ans = 0
        while num != 0:
            ans += num % 10
            num //= 10
        if ans < 10:
            return ans
        else:
            return self.addDigits(ans)


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
